src/pages/_document.tsx

```diff
@@ -2,56 +2,50 @@ import type { DocumentContext, DocumentInitialProps } from 'next/document'
 import NextDocument, { Html, Head, Main, NextScript } from 'next/document'
 import { ServerStyleSheet } from 'styled-components'
 
-class Document extends NextDocument {
-  static async getInitialProps(
-    ctx: DocumentContext
-  ): Promise<DocumentInitialProps> {
-    const sheet = new ServerStyleSheet()
-    const originalRenderPage = ctx.renderPage
+type TDocument = {
+  (): JSX.Element
+  getInitialProps: (ctx: DocumentContext) => Promise<DocumentInitialProps>
+}
```

Comment:
This is a special typing syntax which allow us to type functions that contain properties. Usually, a function would be typed like this:

```ts
type Func = (arg: string) => string
```

But in case the function should contain a property – as functions are objects nonetheless, remember that!? – then we have to use an object typing syntax, and use a special function declaration syntax as the first line within the object type.